twine/commands/upload.py

```diff
@@ -59,15 +76,23 @@ def upload(upload_settings: settings.Settings, dists: List[str]) -> None:
     upload_settings.check_repository_url()
     repository_url = cast(str, upload_settings.repository_config["repository"])
 
+    packages_to_upload = [
+        _make_package(filename, signatures, upload_settings) for filename in uploads
+    ]
+
     print(f"Uploading distributions to {repository_url}")
+    if upload_settings.verbose:
+        for package in packages_to_upload:
+            file_size = utils.get_file_size(package.filename)
+            print(f"  {package.filename} ({file_size})")
+            if upload_settings.sign or package.signed_filename in signatures:
+                print(f"  {package.signed_filename}")
+        print("\n")
```

Comment:
Although I didn't state this, I was hoping that the verbosity would end up in the helper function. Also, I think the signature clause isn't accurate; `package.gpg_signature` seems to be the canonical indication of whether or not a package was signed.

In attempt to understand this and make a concrete suggestion, I went down a rabbit hole and implemented this and other changes in the tests. You can see the diff here:

https://github.com/VikramJayanthi17/twine/compare/testing-add-uploaded-packages-to-verbose...bhrutledge:testing-add-uploaded-packages-to-verbose

To expedite the merge of this PR, I'd like to push my changes to your branch. Is that okay? If you like, I can open a separate PR against this branch on your fork, and we can comment on lines there.

Reply:
That makes sense. Pushing the changes to my branch sounds good, thanks for the help!